semgrep-core/src/core/Guess_lang.ml

```diff
@@ -120,12 +120,7 @@ let split_cmd_re = lazy (Pcre_settings.regexp "[ \t]+")
      "#!/usr/bin/env -S bash -e -u" -> ["/usr/bin/env"; "bash"; "-e"; "-u"]
 *)
 let parse_shebang_line s =
-  let matched =
-    try Some (Pcre.exec ~rex:(Lazy.force shebang_re) s) with
-    | Not_found
-    | Pcre.Error BadUTF8 ->
-        None
-  in
+  let matched = Pcre_settings.exec_noerr ~rex:(Lazy.force shebang_re) s in
```

Comment:
We should augment semgrep.yml and add a rule to say to not use Pcre.exec directly.

Reply:
I don't like the name of the module though Pcre_settings; sounds like it contains PCRE settings when really
we use it as a wrapper over Pcre.